src/components/games/rps.ts

```diff
@@ -0,0 +1,333 @@
+      return 'tie';
+    }
+    if (
+      (player1Sign === RpsGameSign.Paper && player2Sign === RpsGameSign.Rock) ||
+      (player1Sign === RpsGameSign.Scissors && player2Sign === RpsGameSign.Paper) ||
+      (player1Sign === RpsGameSign.Rock && player2Sign === RpsGameSign.Scissors)
+    ) {
+      return 'player1';
+    }
+    return 'player2';
+  }
+
+  public setStatus(timeout: 'player1' | 'player2' | null): void {
+    // Both players submitted a sign
+    if (timeout === null) {
+      /*
+        If one of the players' signs is still pending, something went wrong
+      */
+      if (
+        this.state.player1Sign === RpsGameSign.Pending ||
+        this.state.player2Sign === RpsGameSign.Pending
+      ) {
+        this.state.status = RpsGameStatus.Unknown;
+      } else {
+        const winner = this.determineWinner(this.state.player1Sign, this.state.player2Sign);
+        switch (winner) {
+          case 'player1':
+            this.state.status = RpsGameStatus.Player1Win;
+            break;
+          case 'player2':
+            this.state.status = RpsGameStatus.Player2Win;
+            break;
+          case 'tie':
+            this.state.status = RpsGameStatus.Draw;
+            break;
+          default:
+            this.state.status = RpsGameStatus.Unknown;
+            break;
+        }
+      }
+    } else if (timeout === 'player1') {
+      this.state.status = RpsGameStatus.Player1TimeOut;
+    } else if (timeout === 'player2') {
+      this.state.status = RpsGameStatus.Player2TimeOut;
+    } else {
+      this.state.status = RpsGameStatus.Unknown;
+    }
+  }
+
+  public getEmbedColor(): ColorResolvable {
+    switch (this.state.status) {
+      case RpsGameStatus.Player1Win:
+        return 'GREEN';
+      case RpsGameStatus.Player2Win:
+        return this.state.player2Id ? 'GREEN' : 'RED';
+      case RpsGameStatus.Draw:
+        return 'ORANGE';
+      default:
+        return 'YELLOW';
+    }
+  }
+
+  public getStatusAsString(): string {
+    switch (this.state.status) {
+      case RpsGameStatus.Pending:
+        return 'Game in progress...';
+      case RpsGameStatus.Player1Win:
+        return `${this.state.player1Username} has won, and wins ${
+          this.state.bet
+        } ${getCoinEmoji()} from ${this.state.player2Username}!`;
+      case RpsGameStatus.Player2Win:
+        return `${this.state.player2Username} has won, and wins ${
+          this.state.bet
+        } ${getCoinEmoji()} from ${this.state.player1Username}!`;
+      case RpsGameStatus.Draw:
+        if (!this.state.player2Id) {
+          return `The match ended in a draw, so ${this.state.player2Username} has taken ${
+            this.state.bet / 2
+          } ${getCoinEmoji()} from ${this.state.player1Username}!`;
+        } else {
+          return `The match ended in a draw!`;
+        }
+      // Timeout can be implemented later
+      default:
+        return `Something went wrong! ${getEmojiByName('codeySad')}`;
```

Comment:
This error could also arise if something else went wrong with the game (ie `RpsGameStatus.Unknown`). I don't think we need to mention timeouts, since we don't even check for it yet.